skrobot/planner/sqp_based.py

```diff
@@ -0,0 +1,176 @@
+import numpy as np
+import scipy
+
+try:
+    # in python3
+    from functools import lru_cache
+except ImportError:
+    # in python2
+    from repoze.lru import lru_cache
+
+from skrobot.planner.utils import scipinize
+
+
+def sqp_plan_trajectory(collision_checker,
+                        av_start,
+                        av_goal,
+                        joint_list,
+                        n_wp,
+                        safety_margin=1e-2,
+                        with_base=False,
+                        weights=None,
+                        initial_trajectory=None,
+                        slsqp_option=None
+                        ):
+    """Gradient based trajectory optimization using scipy's SLSQP.
+
+    Collision constraint is considered in an inequality constraits.
+    Terminal constraint (start and end) is considered as an
+    equality constraint.
+
+    Parameters
+    ----------
+    av_start : numpy.array(n_dof,)
+        joint angle vector at start point
+    av_start : numpy.array(n_dof,)
+        joint angle vector at goal point
+    joint_list : list[skrobot.model.Link]
+        link list to be controlled (similar to inverse_kinematics function)
+    n_wp : int
+        number of waypoints
+    safety_margin : float
+        safety margin in collision checking
+    with_base: bool
+        If `with_base=False`, `n_dof` is the number of joints `n_joint`,
+        but if `with_base=True`, `n_dof = len(joint_list) + 3`.
+    weights : Union[numpy.array(n_dof,), None]
+        cost to move of each joint. For example,
+        if you set weights=numpy.array([1.0, 0.1, 0.1]) for a
+        3 DOF manipulator, moving the first joint is with
+        high cost compared to others. If set to `None` it's automatically
+        determined.
+    initial_trajectory : Union[numpy.array(n_wp, n_dof), None]
+        initial solution in the trajectory optimization specifeid by a
+        angle vector sequence. If None, initial trajectory is
+        automatically generated.
+    slsqp_option: Union[Dict, None]
```

Comment:
This `Dict` shoule be `dict.`

Reply:
fixed in https://github.com/iory/scikit-robot/pull/181/commits/25cc3d02aa0309727d461758d0bb62043a73a95e